wavepool/views.py

```diff
@@ -13,9 +13,20 @@ def front_page(request):
             archive: the rest of the newsposts, sorted by most recent
     """
     template = loader.get_template('wavepool/frontpage.html')
-    cover_story = NewsPost.objects.all().order_by('?').first()
-    top_stories = NewsPost.objects.all().order_by('?')[:3]
-    other_stories = NewsPost.objects.all().order_by('?')
+    cover_story = None
+    top_stories = []
+    other_stories = []
+    newsposts = NewsPost.objects.all().order_by('publish_date')
+
+    i = 1
+    for n in newsposts:
+        if n.is_cover_story == True:
+            cover_story = n
+        if i < 3:
```

Comment:
cover_story might be included in top stories here
this could be caught in a unit test

Reply:
if cover story was selected by a filter above there would be no need to check for it here

Reply:
we could just query the first three instead of looping through all of them.
this for loop is probably not necessary if we use query filters instead.

Reply:
noted: existing unit tests were failing - maybe out of scope for junior dev but should look into